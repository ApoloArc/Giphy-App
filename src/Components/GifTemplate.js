import React from 'react'
import PropTypes from 'prop-types'
export const GifTemplate = ({title, url})=> {
    return (
        <div className='card animate__animated animate__fadeInDown'>
            <img src={url} alt={title}/>
            <p className='title-card'>{title}</p>
        </div>
    )
}

GifTemplate.propTypes={
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
}
