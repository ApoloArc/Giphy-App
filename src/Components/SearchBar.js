import React, { useState } from 'react'
import PropTypes from 'prop-types'


export const SearchBar = ({setCategories}) => {

    const [busqueda, setBusqueda] = useState('');
    
    const cambioBusqueda = (e)=>{
        setBusqueda(e.target.value);
    }

    const sinRefresh = e =>{
        e.preventDefault();
        
        setCategories(e=>[busqueda, ...e]);
        setBusqueda('');
    }

    return (
        <form onSubmit={sinRefresh}>
            <input
            type='text'
            placeholder='Busca algo'
            value={busqueda}
            onChange={cambioBusqueda}
            >
            </input>
        </form>
    )
}

SearchBar.propTypes= {
    setCategories: PropTypes.func.isRequired,
}