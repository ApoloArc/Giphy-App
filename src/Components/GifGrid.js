import { GifTemplate } from './GifTemplate';
import { useFetchGifs } from './Hooks/useFetchsGifs';
import PropTypes from 'prop-types';


const GifGrid = ({category}) => {
    
    const {data:images, loading} = useFetchGifs(category);

    return (
        <div>
            <h3 className='titulo-category'>{category}</h3>
            {loading && <p className='animate__animated animate__fadeInDown'>Cargando, espera compare...</p>}
            <div className='grid'>
            {
               images.map(img => <GifTemplate key={img.id} {...img}/>)
            }
            </div>
        </div>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}

export default GifGrid
