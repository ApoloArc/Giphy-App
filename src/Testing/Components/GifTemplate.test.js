import React from 'react'
import {shallow} from 'enzyme'
import { GifTemplate } from '../../Components/GifTemplate'


describe('Debe de mostrar el <GifTemplate/>', () => {

    const title = 'Esto es un t√≠tulo';
    const url = 'https://apolo.com/img.jpg'

    const wrapper = shallow(<GifTemplate title={title} url={url}/>);

    test('Mostrar el componente correctamente', () => {
        
        expect(wrapper).toMatchSnapshot();
    })

    test('debe de tener Animate', () => {
        
        const div = wrapper.find('div');
        const className = div.prop('className');
        expect(className.includes('animate__fadeInDown')).toBe(true);
    })
    

})
