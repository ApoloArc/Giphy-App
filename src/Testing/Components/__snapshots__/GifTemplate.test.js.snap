// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Debe de mostrar el <GifTemplate/> Mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__fadeInDown"
>
  <img
    alt="Esto es un título"
    src="https://apolo.com/img.jpg"
  />
  <p
    className="title-card"
  >
    Esto es un título
  </p>
</div>
`;
